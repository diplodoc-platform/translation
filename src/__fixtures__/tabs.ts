const markdown = `\
{% list tabs %}

- Первый Таб

  Текст первого таба

  - список внутри таба
        
    контент внутри списка
- Второй Таб
  
  Текст второго таба

{% endlist %}

{% list tabs %}

* Первый Таб во второй таб секции

  контент первого таба во второй таб секции

{% endlist %}
`;

const skeleton = `\
{% list tabs %}

- %%%1%%%

  %%%2%%%

  - %%%3%%%
        
    %%%4%%%
- %%%5%%%
  
  %%%6%%%

{% endlist %}

{% list tabs %}

* %%%7%%%

  %%%8%%%

{% endlist %}
`;

const translations = new Map<string, string>([
    ['1', 'First tab'],
    ['2', 'text of the first tab'],
    ['3', 'list inside the tab'],
    ['4', 'content inside the list'],
    ['5', 'Second tab'],
    ['6', 'text of the second tab'],
    ['7', 'First tab inside the second tab section'],
    ['8', 'content of the first tab inside the second tab section'],
]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
