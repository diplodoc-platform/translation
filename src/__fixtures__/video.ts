const markdown = `\
@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) текст после
@[youtube](dQw4w9WgXcQ)
`;

const skeleton = `\
@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) %%%1%%%
@[youtube](dQw4w9WgXcQ)
`;

const translations = new Map<string, string>([['1', 'text after']]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
