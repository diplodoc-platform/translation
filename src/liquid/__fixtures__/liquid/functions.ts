import {Fixtures} from './fixture';

const functions: Fixtures = [
    {
        section: 'functions',
        content: 'Hello P{{ user.name.slice(1, 2) }}vel!',
        number: 1,
    },
    {
        section: 'functions',
        content: 'Sentence {{user.name.slice(1)}} content after.',
        number: 2,
    },
    {
        section: 'functions',
        content: '{{user.slice(1)}}.',
        number: 3,
    },
    {
        section: 'functions',
        content: '{{user_name-cool.slice(1)}}.',
        number: 4,
    },
];

export {functions};
export default {functions};
