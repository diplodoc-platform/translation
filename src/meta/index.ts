function traverse(
    meta: Record<string, unknown>,
    fn: (val: string, path: string[]) => string,
    path = new Array<string>(),
) {
    for (const [key, val] of Object.entries(meta)) {
        const newpath = [...path, key];

        if (typeof val === 'string') {
            // eslint-disable-next-line no-param-reassign
            meta[key] = fn(val, newpath) ?? val;
        } else if (isObjectOrArray(val)) {
            traverse(val, fn, newpath);
        }
    }
}

function isObjectOrArray(obj: unknown): obj is Record<string, unknown> {
    const type = Object.prototype.toString.call(obj);

    return type === '[object Object]' || type === '[object Array]';
}

export {traverse};
export default {traverse};
