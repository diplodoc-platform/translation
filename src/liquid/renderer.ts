import {Token, TokenType} from 'src/liquid';

import {generateX} from 'src/xlf/generator';

export type RendererRuleSet = Record<TokenType | string, RendererRule>;

export type RendererRule = (token: Token) => string;

class Renderer {
    private rendered: string;
    private tokens: Token[];
    private rules: RendererRuleSet;

    constructor(tokens: Token[]) {
        this.tokens = tokens;
        this.rendered = '';

        this.rules = {
            Text: this.plainText.bind(this),
            Space: this.plainText.bind(this),
            If: this.wrapInX.bind(this),
            Else: this.wrapInX.bind(this),
            EndIf: this.wrapInX.bind(this),
            ForInLoop: this.wrapInX.bind(this),
            EndForInLoop: this.wrapInX.bind(this),
            Function: this.wrapInX.bind(this),
            Filter: this.wrapInX.bind(this),
        };
    }

    render() {
        for (const token of this.tokens) {
            const rule = this.rules[token.type];
            if (!rule) {
                throw new Error(`unexpected token: ${token.type}`);
            }

            this.rendered += rule(token);
        }

        return this.rendered;
    }

    private plainText(token: Token) {
        return token.value;
    }

    private wrapInX(token: Token) {
        const {type, value} = token;

        return generateX({
            ctype: `liquid_${type}`,
            equivText: value,
        });
    }
}

export {Renderer};
export default {Renderer};
