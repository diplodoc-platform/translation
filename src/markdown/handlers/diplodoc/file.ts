import {MarkdownRenderer} from '@diplodoc/markdown-it-markdown-renderer';
import Token from 'markdown-it/lib/token';

import {replacer} from 'src/markdown/replacer';

import {MarkdownRendererState} from 'src/markdown/renderer';

function yfmFile(this: MarkdownRenderer<MarkdownRendererState>, tokens: Token[], i: number) {
    const token = tokens[i];
    const content = token.content;

    if (!content?.length) {
        return '';
    }

    token.attrSet('download', replacer(content, this.state));

    return '';
}

export {yfmFile};
export default {yfmFile};
