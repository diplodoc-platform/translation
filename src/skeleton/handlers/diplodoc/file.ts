import {MarkdownRenderer} from '@diplodoc/markdown-it-markdown-renderer';
import Token from 'markdown-it/lib/token';

import {replacer} from 'src/skeleton/replacer';

import {SkeletonRendererState} from 'src/skeleton/renderer';

function yfmFile(this: MarkdownRenderer<SkeletonRendererState>, tokens: Token[], i: number) {
    const token = tokens[i];
    const content = token.content;

    if (!content?.length) {
        return '';
    }

    token.attrSet('download', replacer(content, this.state));

    return '';
}

export {yfmFile};
export default {yfmFile};
