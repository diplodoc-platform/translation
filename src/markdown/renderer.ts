import MarkdownIt from 'markdown-it';
import {
    mdRenderer,
    MarkdownRendererParams,
    MarkdownRendererEnv,
    MarkdownRenderer,
} from '@diplodoc/markdown-it-markdown-renderer';

// configure with diplodoc plugins
// @ts-ignore
import meta from 'markdown-it-meta';
// @ts-ignore
import sup from 'markdown-it-sup';
import notes from '@doc-tools/transform/lib/plugins/notes';
import cut from '@doc-tools/transform/lib/plugins/cut';
import checkbox from '@doc-tools/transform/lib/plugins/checkbox';
import anchors from '@doc-tools/transform/lib/plugins/anchors';
import monospace from '@doc-tools/transform/lib/plugins/monospace';
import imsize from '@doc-tools/transform/lib/plugins/imsize';
import file from '@doc-tools/transform/lib/plugins/file';
import includes from '@doc-tools/transform/lib/plugins/includes';
import tabs from '@doc-tools/transform/lib/plugins/tabs';
import video from '@doc-tools/transform/lib/plugins/video';
import table from '@doc-tools/transform/lib/plugins/table';

import markdownHandlers, {MarkdownHandlersState} from './handlers';
import hooks, {HooksParameters} from './hooks';
import {rules} from './rules';

export type MarkdownRendererState = MarkdownHandlersState;

export type RenderParameters = {
    skeleton: string;
    translations: Map<string, string>;
} & DiplodocParameters;

export type DiplodocParameters = {
    lang?: string;
};

function render(parameters: RenderParameters) {
    if (!validParameters(parameters)) {
        throw new Error('invalid parameters');
    }

    const md = new MarkdownIt({html: true}) as HooksParameters['markdownit'];

    const {handlers, initState} = markdownHandlers.generate(parameters);
    const markdownHooks = hooks.generate({markdownit: md});

    // todo: add non-destructive way of extending markdown-renderer hooks
    for (const defaultHook of Object.entries(MarkdownRenderer.defaultHooks)) {
        const [lifecycle, hooks_] = defaultHook as any;

        if (markdownHooks.hooks[lifecycle]) {
            markdownHooks.hooks[lifecycle] = [...markdownHooks.hooks[lifecycle], ...hooks_];
        }
    }

    const mdOptions: MarkdownRendererParams<MarkdownHandlersState> = {
        handlers,
        initState,
        hooks: markdownHooks.hooks,
        rules,
    };
    const diplodocOptions = {
        lang: parameters.lang ?? 'ru',
        path: '',
    };

    const env: MarkdownRendererEnv = {
        source: parameters.skeleton.split('\n'),
    };

    // diplodoc plugins
    md.use(meta);
    md.use(notes, diplodocOptions);
    md.use(cut, diplodocOptions);
    md.use(sup, diplodocOptions);
    md.use(checkbox, diplodocOptions);
    md.use(anchors, diplodocOptions);
    md.use(monospace, diplodocOptions);
    md.use(imsize, diplodocOptions);
    md.use(file, diplodocOptions);
    md.use(includes, diplodocOptions);
    md.use(tabs, diplodocOptions);
    md.use(video, diplodocOptions);
    md.use(table, diplodocOptions);

    md.use(mdRenderer, mdOptions);

    return md.render(parameters.skeleton, env);
}

function validParameters(parameters: RenderParameters) {
    const {skeleton, translations} = parameters;

    const skeletonCondition = skeleton !== undefined;
    const translationsCondition = translations !== undefined;

    const conditions = [
        skeletonCondition,
        translationsCondition,
        validDiplodocParameters(parameters),
    ];

    return conditions.reduce((a, v) => a && v, true);
}

function validDiplodocParameters(parameters: DiplodocParameters) {
    const {lang} = parameters;

    return lang === undefined || lang === 'ru' || lang === 'en';
}

export {render, validParameters, validDiplodocParameters};
export default {render, validParameters, validDiplodocParameters};
