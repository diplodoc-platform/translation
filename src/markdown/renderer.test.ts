import {RenderParameters, render} from './renderer';

import basic from 'src/__fixtures__/basic';
import variables from 'src/__fixtures__/variables';
import filters from 'src/__fixtures__/filters';
import functions from 'src/__fixtures__/functions';
import conditionals from 'src/__fixtures__/conditionals';
import loops from 'src/__fixtures__/loops';
import meta from 'src/__fixtures__/meta';
import notes from 'src/__fixtures__/notes';
import cuts from 'src/__fixtures__/cuts';
import gfmTables from 'src/__fixtures__/gfm-tables';
import sup from 'src/__fixtures__/sup';
import checkbox from 'src/__fixtures__/checkbox';
import anchors from 'src/__fixtures__/anchors';
import monospace from 'src/__fixtures__/monospace';
import imsize from 'src/__fixtures__/imsize';
import file from 'src/__fixtures__/file';
import links from 'src/__fixtures__/links';
import includes from 'src/__fixtures__/includes';
import strikethrough from 'src/__fixtures__/strikethrough';
import tabs from 'src/__fixtures__/tabs';
import video from 'src/__fixtures__/video';
import multilineTables from 'src/__fixtures__/multiline-tables';

describe('smoke', () => {
    test('it works', () => {
        const parameters = {
            skeleton: '',
            translations: new Map<string, string>(),
        };

        render(parameters);
    });
});

describe('validates parameters', () => {
    it('works with valid parameters', () => {
        const parameters = {
            skeleton: '',
            translations: new Map<string, string>(),
        };

        render(parameters);
    });

    it('throws on invalid parameters', () => {
        const parameters = {} as RenderParameters;
        const invalidLang = {
            skeleton: '',
            translations: new Map<string, string>(),
            lang: 'xx',
        } as RenderParameters;
        expect(() => render(parameters)).toThrow();
        expect(() => render(invalidLang)).toThrow();
    });
});

describe('markdown rendering', () => {
    it('renders identital markdown without translations', () => {
        const parameters = {
            skeleton: '# title\nhello, world\n',
            translations: new Map<string, string>([
                ['1', 'титул'],
                ['2', 'привет, мир'],
            ]),
        };

        const generated = render(parameters);

        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes', () => {
        const parameters = {
            skeleton: basic.skeleton,
            translations: basic.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with liquid variables in text', () => {
        const parameters = {
            skeleton: variables.skeleton,
            translations: variables.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with liquid filters in text', () => {
        const parameters = {
            skeleton: filters.skeleton,
            translations: filters.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with liquid functions in text', () => {
        const parameters = {
            skeleton: functions.skeleton,
            translations: functions.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with liquid conditionals in text', () => {
        const parameters = {
            skeleton: conditionals.skeleton,
            translations: conditionals.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated txt instead of hashes, with liquid loops in text', () => {
        const parameters = {
            skeleton: loops.skeleton,
            translations: loops.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with meta in text', () => {
        const parameters = {
            skeleton: meta.skeleton,
            translations: meta.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with notes in text', () => {
        const parameters = {
            skeleton: notes.skeleton,
            translations: notes.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with cuts in text', () => {
        const parameters = {
            skeleton: cuts.skeleton,
            translations: cuts.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with gfm tables in text', () => {
        const parameters = {
            skeleton: gfmTables.skeleton,
            translations: gfmTables.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with sup syntax in text', () => {
        const parameters = {
            skeleton: sup.skeleton,
            translations: sup.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with checkbox syntax in text', () => {
        const parameters = {
            skeleton: checkbox.skeleton,
            translations: checkbox.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with anchors syntax in text', () => {
        const parameters = {
            skeleton: anchors.skeleton,
            translations: anchors.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with monospace syntax in text', () => {
        const parameters = {
            skeleton: monospace.skeleton,
            translations: monospace.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with imsize syntax in text', () => {
        const parameters = {
            skeleton: imsize.skeleton,
            translations: imsize.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with file syntax in text', () => {
        const parameters = {
            skeleton: file.skeleton,
            translations: file.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with ref links({#T}) syntax in text', () => {
        const parameters = {
            skeleton: links.skeleton,
            translations: links.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, ignoring {% include ... %} syntax in text', () => {
        const parameters = {
            skeleton: includes.skeleton,
            translations: includes.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with striketrough syntax in text', () => {
        const parameters = {
            skeleton: strikethrough.skeleton,
            translations: strikethrough.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with tabs syntax in text', () => {
        const parameters = {
            skeleton: tabs.skeleton,
            translations: tabs.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with video syntax in text', () => {
        const parameters = {
            skeleton: video.skeleton,
            translations: video.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });

    it('renders translated text instead of hashes, with multiline tables syntax in text', () => {
        const parameters = {
            skeleton: multilineTables.skeleton,
            translations: multilineTables.translations,
        };

        const generated = render(parameters);
        expect(generated).toMatchSnapshot();
    });
});
