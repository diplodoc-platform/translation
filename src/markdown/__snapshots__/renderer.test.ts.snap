// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown rendering renders identital markdown without translations 1`] = `
"# title
hello, world"
`;

exports[`markdown rendering renders translated text instead of hashes 1`] = `
"# Heading 1
Paragraph
- First list item
- Second list item
Heading 2
-----------
Paragraph with **bold text** and *italics*

Also key features of the project structure highlight extremely interesting features of the picture as a whole. Sentences add up to paragraphs — and you will enjoy another delirium.
> text inside of the blockquote
1. First ordered list item
2. Second ordered list item

[Link text. Link text continued.](files/file.md \\"Link description. Link description continued.\\")

![Alternative image text Alt. text continued](files/image.png \\"Image title Image title continued\\")"
`;

exports[`markdown rendering renders translated text instead of hashes, ignoring {% include ... %} syntax in text 1`] = `
"{% include notitle [Description](../_includes/file.md) %}

{% include [Description](../_includes/file.md) %}"
`;

exports[`markdown rendering renders translated text instead of hashes, with anchors syntax in text 1`] = `
"# heading with implicit anchor
paragraph
## heading with explicit anchor {#anchor}"
`;

exports[`markdown rendering renders translated text instead of hashes, with checkbox syntax in text 1`] = `
"[ ] first

[x] second
- [_] to be done underscore
- [-] to be done dash
- [ ] to be done space
- [x] done cross
- [X] done capital cross

- list element
  - [x] done
  - [ ] to be done"
`;

exports[`markdown rendering renders translated text instead of hashes, with cuts in text 1`] = `
"
{% cut \\"Cut Title\\" %}

Content inside of the cut

{% endcut %}"
`;

exports[`markdown rendering renders translated text instead of hashes, with file syntax in text 1`] = `"{% file src=\\"sample-file\\" name=\\"file.zip\\" type=\\"application/zip\\" %} text {% file src=\\"sample-file\\" name=\\"file2.zip\\" type=\\"application/zip\\" %}"`;

exports[`markdown rendering renders translated text instead of hashes, with gfm tables in text 1`] = `
"|Header 1|Header 2|
|-----------|-----------|
|[link text](https://www.google.com)|**bold text**|
|![image text](image.png \\"hint\\")|*italics*|
"
`;

exports[`markdown rendering renders translated text instead of hashes, with imsize syntax in text 1`] = `"paragraph with ![image inside](image.png \\"alt text\\" =13x37) with width and height specified"`;

exports[`markdown rendering renders translated text instead of hashes, with liquid conditionals in text 1`] = `
"# Титул Документа
{% if OS == 'iOS' %}

Скачайте приложение из  [{{ ios-marketplace }} Стора](https://www.apple.com/ios/app-store).

{% else %}

Скачайте приложение из  [{{ other-marketplace }} Стора](https://play.google.com).

Какой-то текст {% if  OS == 'iOS' %} Apple {% else %} Android {% endif %} Продолжение текста.

{% endif %}"
`;

exports[`markdown rendering renders translated text instead of hashes, with liquid filters in text 1`] = `
"Титул Документа
Привет {{ user.name | capitalize }}!

всего пользователей: {{ users | length }}"
`;

exports[`markdown rendering renders translated text instead of hashes, with liquid functions in text 1`] = `
"Титул Документа
Привет П{{ user.name.slice(1) }}!

Привет П{{ user.name.slice(1, 2) }}вел!"
`;

exports[`markdown rendering renders translated text instead of hashes, with liquid variables in text 1`] = `
"# {{ service-name }} Сервис
Попробуйте {{ service-name }} и расскажите нам что вы думаете о нём.

Также присмотритесь к [{{ other-service-name }}](files/file.md \\"{{ other-service-name }} описание\\")

![скриншот использования {{ service-name }}](files/image.png \\"{{ service-name }} в действие\\")"
`;

exports[`markdown rendering renders translated text instead of hashes, with meta in text 1`] = `
"---
title: Creation of the Windows virtual machine
description: >-
  Create Windows virtual machine with {{ compute-short-name }} service. You can
  connect to the virtual machine using Remote Desktop Protocol (RDP). Before
  connecting ensure that NLA is enabled.
key: |
  Sammy Sosa

  Let's Goooo
seq:
  - windows
  - windows vm
  - windows vm
  - windows virtual machine
  - virtual machine
  - vm
seq_of_mappings:
  - name: Mark
    age: 13
  - name: Noam
    age: 99
object_inline:
  name: John Smith
  age: 33
array_inline:
  - milk
  - pumpkin pie
  - eggs
  - juice
seq_of_seq:
  - - Mark McGwire
    - 65
    - 0.278
  - - Sammy Sosa
    - 63
    - 0.288
nesting:
  - children:
      - name: bob
      - name: marry
---
# Meta
Comments and metadata - not displayed in the final file. They are used to make comments for other writers or set meta into final html for seo."
`;

exports[`markdown rendering renders translated text instead of hashes, with monospace syntax in text 1`] = `
"##monospace##
this ##is## mono**space** he##he##he
[hello ##world## i'm ##fine##](href \\"hint\\")"
`;

exports[`markdown rendering renders translated text instead of hashes, with multiline tables syntax in text 1`] = `
"#|
||
**First Heading**
|
**Second Heading**
||
||
first cell text
|
second cell text
||
|#

- text before table

  #|
  ||
  First Heading
  |
  Second Heading
  ||
  ||
  first cell text
  |
  second cell text
  ||
  |#

  text after table"
`;

exports[`markdown rendering renders translated text instead of hashes, with notes in text 1`] = `
"
{% note info \\"Information\\" %}

information

{% endnote %}
- list with note inside

  {% note alert \\"Alert\\" %}

  alert

  {% endnote %}"
`;

exports[`markdown rendering renders translated text instead of hashes, with ref links({#T}) syntax in text 1`] = `"paragraph with ref link [{#T}](file.md \\"title\\") paragraph continuation"`;

exports[`markdown rendering renders translated text instead of hashes, with striketrough syntax in text 1`] = `
"~~striketrough~~
[hello ~~world~~ i'm ~~fine~~](href \\"hint\\")"
`;

exports[`markdown rendering renders translated text instead of hashes, with sup syntax in text 1`] = `
"paragraph of text 1^st^ line
this is 2^nd^ line
[1^st^ link inside the paragraph](file.md)"
`;

exports[`markdown rendering renders translated text instead of hashes, with tabs syntax in text 1`] = `
"{% list tabs %}

- First tab

  text of the first tab
  - list inside the tab
    
    content inside the list
- Second tab

  text of the second tab

{% endlist %}
"
`;

exports[`markdown rendering renders translated text instead of hashes, with video syntax in text 1`] = `
"@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) text after
@[youtube](dQw4w9WgXcQ)"
`;

exports[`markdown rendering renders translated txt instead of hashes, with liquid loops in text 1`] = `
"# Титул Документа
{% for element in array %}

Какой-то текст {{ variable_name }} продолжение текста.

{% endfor %}

Перед однострочным циклом {% for element in array %} текст внутри содержащий {{variable}} {% endfor %} после однострочного цикла"
`;
